### PR TITLE
Update sample app withe the latest changes to compiler

### PR DESCRIPTION
**Pull request by sav007**

### CODE REVIEW

File: apollo-sample/src/main/java/com/example/apollographql/sample/SampleApplication.java [hunk 1]
```diff
@@ -31,10 +32,12 @@
         .baseUrl(BASE_URL)
         .client(okHttpClient)
         .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
-        .addConverterFactory(new ApolloConverterFactory.Builder().build())
+        .addConverterFactory(new ApolloConverterFactory.Builder()
+            .withResponseFieldMapper(FeedQuery.Data.class, new FeedQuery.Data.Mapper(FeedQuery.Data.FACTORY))
```
Comment (digitalbuddha) [inline apollo-sample/src/main/java/com/example/apollographql/sample/SampleApplication.java]: so sorry could you explain this line again.  Will we need to register every object or is this for a particular custom field

Comment (sav007) [inline apollo-sample/src/main/java/com/example/apollographql/sample/SampleApplication.java]: Yes user must provide Mapper instance for each query top level Data class. Converter then will use it to parse query response. So basically number of queries == number of mapper registrations

Comment (sav007) [inline apollo-sample/src/main/java/com/example/apollographql/sample/SampleApplication.java]: We will need to update readme in next release

**Comment by digitalbuddha** [comment]
thanks lgtm!

**Pull request merged**